src/targets/services/timeseriesWithoutAggregateMigration.js

```diff
@@ -27,7 +27,9 @@ const migrateTimeSeriesWithoutAggregation = async () => {
   const migratedTimeseries = computeAggregatedTimeseries(resp.data)
 
   // Save the migrated timeseries
-  await client.saveAll(migratedTimeseries)
+  for (const timeserie of migratedTimeseries) {
```

Comment:
nit: désolé je vais pas être super précis, mais je sais qu'il y a une différence en JS entre les différences manières de boucler et de lancer des async dans la boucle. Certaines vont tout exécuter en parallèle, et d'autres en serie (diff entre le for et le forEach peut-être ?)

Voilà pour info si jamais ici on veut spécifiquement du série ou du parallèle, à vérifier si on utilise la bonne manière de boucler...

Reply:
nit: je le dis parce que je le pense très fort... on devrait / on aurait du ajouter des tests concernant les dernières modifs qu'on a fait sur le service...

Reply:
Oui, on avait fait un zoom ensemble sur le sujet ;) Si tu fais un forEach, le await ne fonctionnera pas et les promesses seront lancées en //
Avec un for... of, le await de la promesse sera bien respecté, c'est ce qu'on veut ici

Reply:
> Saving in bulk is way much performant, but might occasionally fail
without any info. To prevent this, we now save the docs individually.

I kinda don't like that. It can be interesting to get more informations about that. We use bulk_save in several places and if we find out that it's not working correctly, we need to change the way we use it everywhere. 

Do you have more informations to share about that specific case? Reproductible bug or something? Do you have the bug only on Couch 2.3 or also on 3.2.2 ? 

Maybe we can add some logs to cozy-client when using bulk_save? Do you have asked @clochix to check the logs to see what happened? 

(sorry for the few questions)

Reply:
Pour les tests, je parle en général. On a passé quelques fixes dernièrement autours de ce sujet, sans pour autant rajouter de test. Mon sentiment est peut-être biaisé car je ne fais que de la review, mais ça me conforte dans mon idée initiale qu'on aurait pu en rajouter.

Reply:
@Crash-- I don't like that either :/
From what I saw, when at least 1 doc fails to be saved for any reason, it ends up with a non-helping error `Error while bulk saving`. And I couldn't reproduced this error, that sometimes happens... For now, I only tested on couch 2.3, I don't know if the 3.2 improves this somewhat. We should indeed try to find a reproductible scenario and improve the error, if it's possible.

I would be interesting to know if this error happens elsewhere, @clochix is it possible to spot `Error while bulk saving` on our logs among all the instances?